management/WFCORE-1110_Custom_HTTP_Management_Headers.adoc

```diff
@@ -0,0 +1,139 @@
+ * This will be a new `constant-headers` attribute on the `/core-service=management/management-interface=http-interface` resource.
+ * Within the `constant-headers` attribute it will be possible to specify a set of headers using key / value pairs against a specific path.
+ * All paths will be assumed to be prefixes starting with `/` - any paths missing a leading `/` will be handled as though the leading `/` has been specified.
+ * If a specific header is defined multiple times it will be added to the response multiple times.
+ * Configured headers will not override any headers added by the corresponding endpoint.
+
+
+The new attribute to be added to the `/core-service=management/management-interface=http-interface` resource will be called `constant-headers` and will be defined as: -
+
+----
+            "constant-headers" => {
+                "type" => LIST,
+                "description" => "The set of constant HTTP headers to apply to response messages.",
+                "expressions-allowed" => false,
+                "required" => false,
+                "value-type" => {
+                    "path" => {
+                        "type" => STRING,
+                        "description" => "The path the headers should be applied to.",
+                        "expressions-allowed" => true,
+                        "required" => true,
+                    },
+                    "headers" => {
+                        "type" => LIST,
+                        "description" => "The headers to set for the matched path.",
+                        "expressions-allowed" => false,
+                        "required" => false,
+                        "value-type" => {
+                            "name" => {
+                                "type" => STRING,
+                                "description" => "The name of the HTTP header to set.",
+                                "expressions-allowed" => true,
+                                "required" => true,
+                            },
+                            "value" => {
+                                "type" => STRING,
+                                "description" => "The value to set for the HTTP header.",
+                                "expressions-allowed" => true,
+                                "required" => true,
+                            }
+                        }
+                    }
+                }
+----
+
+The following example hows how the interface can be configured to add a constant header for the `/management` endpoint: -
+
+----
+/] /core-service=management/management-interface=http-interface:write-attribute(name=constant-headers, value=[{path=/management, headers=[{name=X-Header, value=HeaderValue}]}])
+----
+
+The `write-attribute` operation will trigger a `reload-required` state as the management interface will need to be re-initialised.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
```

Comment:
I think we should mention here that the server won't give any special meaning to added headers and that this feature isn't meant to configure response content. I mean for example a situation when user adds constant header {Content-Type: application/yaml}, they can't expect management port to return content in yaml.